RIOT-OS/RIOT
# matrix_keypad: Add matrix-style keypad module

### Contribution description

This module implements a simple matrix keypad driver where keys are connected 
between GPIO columns and rows. It works best with diodes in series with the   
switches to prevent key ghosting, but it can be used without.

For example one of these: https://learn.adafruit.com/matrix-keypad        

### Testing procedure

Test output with a custom board:
```
2022-10-13 13:54:23,675 # SEGGER J-Link V7.58 - Real time terminal output
2022-10-13 13:54:23,676 # J-Link EDU Mini V1 compiled Nov  2 2021 11:12:01 V1.0, SN=801025014
2022-10-13 13:54:23,676 # Process: JLinkExe
2022-10-13 13:54:23,676 # main(): This is RIOT! (Version: 2022.10-devel-1128-gfa62a-pr/driver/matrix_keypad)
2022-10-13 13:54:23,676 # Generated RIOT application: 'matrix_keypad'
2022-10-13 13:54:23,677 # [OK]
2022-10-13 13:54:23,677 #
2022-10-13 13:54:24,242 # Key switch at column 2 and row 1 is pressed!
2022-10-13 13:54:24,394 # Key switch at column 2 and row 1 is released!
2022-10-13 13:54:24,923 # Key switch at column 2 and row 2 is pressed!
2022-10-13 13:54:25,037 # Key switch at column 2 and row 2 is released!
2022-10-13 13:54:25,294 # Key switch at column 2 and row 3 is pressed!
2022-10-13 13:54:25,395 # Key switch at column 2 and row 3 is released!
2022-10-13 13:54:25,611 # Key switch at column 2 and row 4 is pressed!
2022-10-13 13:54:25,712 # Key switch at column 2 and row 4 is released!
2022-10-13 13:54:25,946 # Key switch at column 1 and row 3 is pressed!
2022-10-13 13:54:26,047 # Key switch at column 1 and row 3 is released!
```

Full Kconfig configuration is provided with the driver (to the best of my ability), I would love it if somebody could sanity check it.

### Issues/PRs references

None
